community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -446,7 +446,7 @@ public void close() throws TransactionFailureException
         closing = true;
         try
         {
-            if ( failure || !success )
+            if ( failure || !success || terminated )
```

Comment:
We set `failure` flag to `true` in `#markForTermination()` as well as `terminated`. Maybe it is ok to just keep this check as is?